seldonite/analyze.py

```diff
@@ -59,4 +59,41 @@ def countries(x):
                 df_with_key = df.withColumn("key", lit(key))
 
                 # TODO: what should I return?
-                key.groupBy(col("key")).count().show()
\ No newline at end of file
+                key.groupBy(col("key")).count().show()
+        
+
+        def keywords_over_time_delal(self, year):
+            self.collector._check_args()
+
+            spark_builder = self.collector._get_spark_builder()
+
+            with spark_builder.start_session() as spark_manager:
+                df = self.collector._fetch(spark_manager)
+
+                df = df.withColumn('all_text', psql.functions.concat(df['title'], psql.functions.lit(' '), df['text']))
+
+                year_date = year(col("date")).alias("date_year")
+                month_date = month(col("date")).alias("date_month")
+                sparkDF = sparkDF.withColumn('date_year', lit(year_date))
+                sparkDF = sparkDF.withColumn('date_month', lit(month_date))
+
+                d = {}
+
+                for i in keywords:
+                    d[i] = 0
+
+                def count_keywords(x):
+                    # key events
+                    keywords = ["Trump"]
+                    
+                    for event in keywords:
+                        if event in x:
+                            
+                            d[event] += 1
+                    return d
+
+                udfKeywords = udf(count_keywords, MapType(StringType(), IntegerType()))
+                df = sparkDF.withColumn('counts', udfKeywords(sparkDF.all_text))
```

Comment:
Instead of using a udf here which is fairly slow, try to use built in spark functions. This code will be much faster if you do something like this:
```
for keyword in keywords:
    df = df.withColumn(keyword, sfuncs.col('all_text').like(f"%{keyword}%"))

df = df.groupby('date_year', 'date_month').count(*keywords)